### PR TITLE
User Telemetry & APM with Datadog (Review by Gavin)

### PR DESCRIPTION
**Pull request by steven2000nz**
User Telemetry & APM with Datadog

### CODE REVIEW

File: docs/HOL-Error_Reporting_and_Monitoring_with_DataDog/README.md [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+    {
+        ...
+
+        services.AddScoped<IDataDogSettings>(p => new ConfigurationDataDogSettings(Configuration.GetSection(ConfigurationPath.Combine("Keys", "DataDog"))));
+
+        ...
+    }
+
+
+**Step 5.** Awesome. Now we want to create the contract we're using to communicate with DataDog. This can be found in the DataDog API -> http://docs.datadoghq.com/api/?lang=console#events. Create a new class called DataDogEventRequest.cs
```
Comment (gavinbarron) [inline docs/HOL-Error_Reporting_and_Monitoring_with_DataDog/README.md]: Use the markdown link format [name](url) please

File: docs/HOL-Error_Reporting_and_Monitoring_with_DataDog/README.md [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+            SignInManager = signInManager;
+        }
+
+        ...
+
+    }
+
+Then in the login method we want to trace when there's a failed login attempt. Just underneath ModelState.AddModelError we want to trace the error and log it to DataDog.
+
+     //
```
Comment (gavinbarron) [inline docs/HOL-Error_Reporting_and_Monitoring_with_DataDog/README.md]: Should these comment lines be here?

File: docs/HOL-Error_Reporting_and_Monitoring_with_DataDog/README.md [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+
+![](<media/failed-logins.png>)
+
+Give it some time then check your emails. You should have an alert sitting in your inbox.
+
+![](<media/email.png>)
+
+You will also get a follow up email if the event stops triggering after a set time period. You can leverage this to ensure you don't have any false positives.
+
+![](<media/recovered.png>)
```
Comment (gavinbarron) [inline docs/HOL-Error_Reporting_and_Monitoring_with_DataDog/README.md]: Needs a short wrap up

File: docs/HOL-Error_Reporting_and_Monitoring_with_DataDog/README.md (hunk 1 above)
Comment (bernarden) [inline docs/HOL-Error_Reporting_and_Monitoring_with_DataDog/README.md]: Done

File: docs/HOL-Error_Reporting_and_Monitoring_with_DataDog/README.md [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+    public class DataDogEventRequest
+    {
+        public string Title { get; set; }
+        public string Text { get; set; }
+        public string Priority { get; set; }
+        public List<string> Tags { get; set; }
+        public string Alert_Type { get; set; }
+    }
+
+**Step 6.** Add the Microsoft.AspNet.WebApi.Client nuget package. We need this to perfrom the PostAsJsonAsync method on our HttpClient
```
Comment (bernarden) [inline docs/HOL-Error_Reporting_and_Monitoring_with_DataDog/README.md]: It was a misleading screenshot. The package wasn't actually pre-release. 
Updated the screenshot.

File: docs/HOL-Error_Reporting_and_Monitoring_with_DataDog/README.md (hunk 2 above)
Comment (bernarden) [inline docs/HOL-Error_Reporting_and_Monitoring_with_DataDog/README.md]: Done

File: docs/HOL-Error_Reporting_and_Monitoring_with_DataDog/README.md [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+
+Now we want to create a publish profile to a Microsoft Azure App Service.
+
+![](<media/publish1.png>)
+
+If you have an app service created in Azure, you can select it from here. Otherwise you can select 'New...' to create one.
+
+![](<media/publish2.png>)
+
+Create a unique web app name and assign it to a familar resource group. Note: Keep similar items in the same resource group. As your Azure subscription grows this will assist greatly for knowing which things are related.
```
Comment (bernarden) [inline docs/HOL-Error_Reporting_and_Monitoring_with_DataDog/README.md]: Done

**Pull request closed without merging**